pycargoebuild/__main__.py

```diff
@@ -71,13 +72,26 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def get_cargo_lock_file(directory: Path) -> io.BufferedReader:
+        root = directory.absolute().root
+        current_dir = directory
+        while True:
+            try:
+                return open(current_dir / "Cargo.lock", "rb")
+            except FileNotFoundError as e:
+                err = e
+            if current_dir.samefile(root):
+                break
+            current_dir = current_dir / ".."
```

Comment:
Actually, thinking about it, it probably makes more sense to preserve the first error, i.e. the one with the original path.

```suggestion
        try:
            return open(directory / "Cargo.lock", "rb")
        except FileNotFoundError as e:
            err = e
        directory /= ".."
        while not directory.samefile(root):
            try:
                return open(directory / "Cargo.lock", "rb")
            except FileNotFoundError as e:
                pass
            directory /= ".."
```

The next logical step would be to add a generator function that appends `../` until it hits root, and then do something like:

```
for directory in iterate_parents(directory, root): ...
```

If you prefer that I did it, please just let me know.